src/http/authentication/authentication_types.h

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "enclave/rpc_context.h"
+#include "kv/tx.h"
+
+#include <memory>
+
+namespace ccf
+{
+  struct AuthnIdentity
```

Comment:
I didn't feel like wearing out my keyboard writing `Authentication` everywhere, but wanted to distinguish the ancient enemies AuthN and AuthZ. So this and similar types get a little `n`. But some variable names use the full `authentication`, because I hate consistency. Leaning towards replacing `authn` everywhere with `authentication`, unless anyone objects to the verbosity?

Reply:
Fine with me, I think AuthN (as opposed to Authn) would have been quite acceptable too.